src/types.ts

```diff
@@ -6,12 +6,39 @@ import type { Index } from "@upstash/vector";
 import type { CustomPrompt } from "./rag-chat";
 import type { ChatMistralAI } from "@langchain/mistralai";
 import type { ChatAnthropic } from "@langchain/anthropic";
+import type { CoreTool } from "ai";
 
 declare const __brand: unique symbol;
 type Brand<B> = { [__brand]: B };
 export type Branded<T, B> = T & Brand<B>;
 type OptionalAsync<T> = T | Promise<T>;
 
+export type ToolingOptions = {
+  /**
+   * Tools to use when calling the LLM.
+   * https://sdk.vercel.ai/docs/ai-sdk-core/tools-and-tool-calling
+   */
+  tools?: Record<string, CoreTool>;
+  /**
+   * Maximum number of steps to take when using tools.
+   * @default 5
+   */
+  maxSteps?: number;
+  /**
+   * The tool choice strategy
+   * Whether to force the model to attempt tool usage.
+   * @default 'auto'
+   */
+  toolChoice?:
+    | "auto"
+    | "none"
+    | "required"
+    | {
+        type: "tool";
+        toolName: keyof Record<string, CoreTool>;
+      };
+};
```

Comment:
I'm not adding all the tooling support, just the basics, in case you can ping me, that I can add more options